File: APE/SW-AyLP-APE-02: Estructuras de Selección/Códigos_Java/gestionJean.java
import java.util.Scanner;

public class gestionJean {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String codigo = "";
        String color = "";
        String talla = "";
        boolean fueTenido = false;
        int numTenidos = 0;
        double precio = 0.0;
        int numBotones = 0;
        int humedad = 0;
        String estadoTela = "";
        int opcion = 0;

        do {
            System.out.println("\n--- MENU PRINCIPAL ---");
            System.out.println("1. Registrar datos del jean");
            System.out.println("2. Mostrar datos");
            System.out.println("3. Lavar jean");
            System.out.println("4. Secar jean");
            System.out.println("5. Salir");
            System.out.print("Elija una opcion: ");
            opcion = scanner.nextInt();
            scanner.nextLine();

            switch (opcion) {
                case 1:
                    System.out.print("Ingrese codigo: ");
                    codigo = scanner.nextLine();

                    System.out.print("Ingrese color: ");
                    color = scanner.nextLine();

                    System.out.print("Ingrese talla: ");
                    talla = scanner.nextLine();

                    System.out.print("¿Fue tenido? (true/false): ");
                    fueTenido = scanner.nextBoolean();

                    System.out.print("Número de tenidos: ");
                    numTenidos = scanner.nextInt();

                    System.out.print("Precio: ");
                    precio = scanner.nextDouble();

                    System.out.print("Numero de botones: ");
                    numBotones = scanner.nextInt();

                    System.out.print("Humedad (%): ");
                    humedad = scanner.nextInt();
                    scanner.nextLine();

                    System.out.print("Estado de la tela: ");
                    estadoTela = scanner.nextLine();
                    break;

                case 2:
                    System.out.println("\n--- DATOS DEL JEAN ---");
                    System.out.println("Codigo: " + codigo);
                    System.out.println("Color: " + color);
                    System.out.println("Talla: " + talla);
                    System.out.println("Fue tenido: " + fueTenido);
                    System.out.println("Número de tenidos: " + numTenidos);
                    System.out.println("Precio: " + precio);
                    System.out.println("Numero de botones: " + numBotones);
                    System.out.println("Humedad: " + humedad + "%");
                    System.out.println("Estado de la tela: " + estadoTela);
                    break;

                case 3:
                    if (numTenidos > 0) {
                        numTenidos--;
                        System.out.println("El jean ha sido lavado. Tenidos restantes: " + numTenidos);
                    } else {
                        System.out.println("El jean no tiene tenidos que reducir.");
                    }
                    break;

                case 4:
                    if (humedad > 0) {
                        humedad -= 10;
                        if (humedad < 0) {
                            humedad = 0;
                        }
                        System.out.println("El jean ha sido secado. Humedad actual: " + humedad + "%");
                    } else {
                        System.out.println("El jean ya esta seco.");
                    }
                    break;

                case 5:
                    System.out.println("Saliendo del sistema...");
                    break;

                default:
                    System.out.println("Opcion invalida.");
                    break;
            }
        } while (opcion != 5);

        scanner.close();
    }
}

File: APE/SW-AyLP-APE-02: Estructuras de Selección/Códigos_C++/gestionJean.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    // Variables independientes para reemplazar el Registro
    string codigo = "";
    string color = "";
    string talla = "";
    bool fueTenido = false;
    int numTenidos = 0;
    double precio = 0.0;
    int numBotones = 0;
    int humedad = 0;
    string estadoTela = "";

    int opcion = 0;
    char respuestaTenido; // Para capturar V/F del usuario

    do {
        cout << "\n--- MENÚ PRINCIPAL ---" << endl;
        cout << "1. Registrar datos del jean" << endl;
        cout << "2. Mostrar datos" << endl;
        cout << "3. Lavar jean" << endl;
        cout << "4. Secar jean" << endl;
        cout << "5. Salir" << endl;
        cout << "Elija una opción: ";
        cin >> opcion;

        switch (opcion) {
            case 1:
                cout << "Ingrese código: ";
                cin >> codigo;

                cout << "Ingrese color: ";
                cin >> color;

                cout << "Ingrese talla: ";
                cin >> talla;

                cout << "¿Fue teñido? (V/F): ";
                cin >> respuestaTenido;
                fueTenido = (respuestaTenido == 'V' || respuestaTenido == 'v');

                cout << "Número de teñidos: ";
                cin >> numTenidos;

                cout << "Precio: ";
                cin >> precio;

                cout << "Número de botones: ";
                cin >> numBotones;

                cout << "Humedad (%): ";
                cin >> humedad;

                cout << "Estado de la tela: ";
                cin >> estadoTela;
                break;

            case 2:
                cout << "\n--- DATOS DEL JEAN ---" << endl;
                cout << "Código: " << codigo << endl;
                cout << "Color: " << color << endl;
                cout << "Talla: " << talla << endl;
                cout << "Fue teñido: " << (fueTenido ? "Sí" : "No") << endl;
                cout << "Número de teñidos: " << numTenidos << endl;
                cout << "Precio: " << precio << endl;
                cout << "Número de botones: " << numBotones << endl;
                cout << "Humedad: " << humedad << "%" << endl;
                cout << "Estado de la tela: " << estadoTela << endl;
                break;

            case 3:
                if (numTenidos > 0) {
                    numTenidos--;
                    cout << "El jean ha sido lavado. Teñidos restantes: " << numTenidos << endl;
                } else {
                    cout << "El jean no tiene teñidos que reducir." << endl;
                }
                break;

            case 4:
                if (humedad > 0) {
                    humedad -= 10;
                    if (humedad < 0) {
                        humedad = 0;
                    }
                    cout << "El jean ha sido secado. Humedad actual: " << humedad << "%" << endl;
                } else {
                    cout << "El jean ya está seco." << endl;
                }
                break;

            case 5:
                cout << "Saliendo del sistema..." << endl;
                break;

            default:
                cout << "Opción inválida." << endl;
                break;
        }
    } while (opcion != 5);

    return 0;
}